Data/Aeson/Types/Internal.hs

```diff
@@ -342,6 +343,11 @@ instance Monoid (Parser a) where
     mappend = (<>)
     {-# INLINE mappend #-}
 
+-- | Raise a parsing failure with some custom message.
+parseFail :: String -> Parser a
+parseFail = fail
+{-# INLINE parseFail #-}
```

Comment:
`{-# INLINE #-}` is IMO not necessary. I'm quite sure GHC does right thing here. (`aeson` have too many `INLINE`s; we should really remove probably most of them).

Reply:
Haha I added the INLINE precisely because I thought I'd be yelled at for not! ;)

Fixed now.